include/deal.II/grid/grid_tools.h

```diff
@@ -461,6 +461,35 @@ namespace GridTools
                       const double max_ratio = 1.6180339887,
                       const unsigned int max_iterations = 5);
 
+  /**
+   * Analyze the boundary cells of a mesh, and if one cell is found at
+   * a corner position (with dim adjacent faces on the boundary), and its
+   * dim-dimensional angle fraction exceeds @p limit_angle_fraction,
+   * refine globally once, and replace the children of such cell
+   * with children where the corner is no longer offending the given angle
+   * fraction.
+   *
+   * If no boundary cells exist with two adjacent faces on the boundary, then
+   * the triangulation is left untouched. If instead we do have cells with dim
+   * adjacent faces on the boundary, then the fraction between the dim-dimensional
+   * solid angle and dim*pi/2 is checked against the parameter @p limit_angle_fraction.
+   * If it is higher, the grid is refined once, and the children of the
+   * offending cell are replaced with some cells that instead respect the limit. After
+   * this process the triangulation is flattened, and all Manifold objects are restored
+   * as they were in the original triangulation.
+   *
+   * @param[in,out] tria Triangulation to regularize.
+   *
+   * @param[in] limit_angle_fraction Maximum ratio of angle or solid
+   * angle that is allowed for a corner element in the mesh.
+   *
+   * @author Luca Heltai, Martin Kronbichler, 2017
+   */
+  template <int dim, int spacedim>
+  void
+  regularize_mesh(Triangulation<dim,spacedim> &tria,
```

Comment:
maybe `regularize_corner_cells` is a better name as one could think of some other algorithms which regularize the whole mesh.